soroban-sdk-macros/src/derive_enum.rs

```diff
@@ -378,7 +378,7 @@ fn map_tuple_variant(
         ScSpecUdtUnionCaseV0::TupleV0(ScSpecUdtUnionCaseTupleV0 {
             doc: docs_from_attrs(attrs),
             name: case_name.try_into().unwrap_or_else(|_| StringM::default()),
-            type_: field_types.try_into().unwrap(),
+            type_: field_types,
```

Comment:
The type is already the same type as the destination, so the conversion is redundant.